pkg/apis/intevents/v1alpha1/brokercell_lifecycle.go

```diff
@@ -87,6 +87,10 @@ func (bs *BrokerCellStatus) MarkIngressFailed(reason, format string, args ...int
 	brokerCellCondSet.Manage(bs).MarkFalse(BrokerCellConditionIngress, reason, format, args...)
 }
 
+func (bs *BrokerCellStatus) MarkIngressUnknown(reason, format string, args ...interface{}) {
```

Comment:
When it is unknown varies in different scenarios, there is no strict rules/direct guidelines for that (the only documentation is https://knative.dev/docs/serving/spec/knative-api-specification-1.0/#error-signalling, indicating unknown represents reconciliation is not yet complete and success or failure is not yet determined, which is self-explained). I guess this might be the reason why In XX_lifecycle.go files, we don't have the convention to add more explanation for markUnkown/markFailed/markReady function.

Reply:
What do you think about adding "Read https://knative.dev/docs/serving/spec/knative-api-specification-1.0/#error-signalling" as GoDoc? I would have found it useful, so I think other newcomers would as well.

Reply:
I am ok to add it.

Reply:
On second thought, I think this link is too generic for this specific function, it is talking about all the things about status, and for marking rules about unknown status, it only mentioned `unknown represents reconciliation is not yet complete and success or failure is not yet determined`.

It is odd to add this comment only to this MarkUnkown function as this link contains information about all MarkTrue/MarkFalse/MarkUnknown functions, and not really related on the markUnknown function itself. However,1) if we add this comment to all those Mark functions, that is redundant. 2)If we only add comment to this one, as we have so many MarkUnknown function in our code base, it is unlikely the newcomers will see exactly this one and find the information.

Reply:
If you would like to learn more about condition and status, one more proper way I think is to add a doc about marking status under https://github.com/google/knative-gcp/tree/master/docs/spec, like serving did in their repo https://github.com/knative/serving/blob/master/docs/spec/errors.md

Reply:
Then we can add https://knative.dev/docs/serving/spec/knative-api-specification-1.0/#error-signalling together with more specific examples like https://github.com/knative/eventing/issues/2187#issuecomment-562283838 to that doc, newcomer can refer to that doc to find more information about marking rules.

Reply:
Synced offline, https://github.com/google/knative-gcp/issues/1959 created for adding such doc